parsec/core/invite_claim.py

```diff
@@ -64,8 +64,14 @@ class InviteClaimInvalidTokenError(InviteClaimError):
     pass
 
 
-def generate_invitation_token():
-    return token_hex(8)
+def generate_invitation_token(length=6):
+    """Generate a token of 6 digits, whith the first one different than 0.
+
+    Equivalent of `randint(100000, 999999)`.
+    """
+    mini = int("1".ljust(length, "0"))
+    maxi = int("9".ljust(length, "9"))
+    return str(secrets.choice(range(mini, maxi + 1)))
```

Comment:
Perso j'aurai plutôt écrit ça sous la forme:
```python
''.join(str(secrets.randbelow(10)) for _ in range(length))
```
l'avantage étant qu'on voit clairement l'utilisation length dans la boucle for.

En plus de ça j'aurai eu tendance à penser que `secrets.choice` est obliger de transformer l'itérateur en liste avant de choisir une valeur dedans (du coup `secrets.choice(range(mini, maxi + 1))` nous fait créer une liste de 900000 éléments...

Mais au final les faits me donne tord:
```shell
$ python -m timeit 'from secrets import choice; length = 6; mini = int("1".ljust(length, "0")); maxi = int("9".ljust(length, "9"))' 'str(choice(range(mini, maxi + 1)))'
100000 loops, best of 3: 5.67 usec per loop
$ python -m timeit 'from secrets import randbelow; length = 6' '"".join(str(randbelow(10)) for _ in range(length))'
10000 loops, best of 3: 32.4 usec per loop
$ python -m timeit 'from secrets import randbelow; length = 6' 'str(randbelow(10 * length)).ljust(length, "0")'
100000 loops, best of 3: 4.51 usec per loop
```

J'imagine que l'appel à `secrets.randbelow` est couteux (ce qui est confirmé en testant avec `length = 12` vu qu'on obtient un temps 2 fois plus long avec ma solution contre grosso modo le même temps avec ta solution)

Du coup j'ai fait une 3ème solution qui combine plus ou moins nos 2 solutions:
```python
str(secrets.randbelow(10 * length)).ljust(length, "0")
```
l'avantage de cette solution est qu'on a l'usage de length qui est toujours plutôt simple et qu'en plus on évite le soucis de valeur minimal qu'il y a dans ta solution et dont je parlais plus haut.

Enfin un dernier soucis qu'il y a dans ta solution (et dans ma dernière solution):
```
python -m timeit 'from secrets import choice; length = 20; mini = int("1".ljust(length, "0")); maxi = int("9".ljust(length, "9"))' 'str(choice(range(mini, maxi + 1)))'
Traceback (most recent call last):
  File "/usr/lib/python3.6/timeit.py", line 324, in main
    number, _ = t.autorange(callback)
  File "/usr/lib/python3.6/timeit.py", line 223, in autorange
    time_taken = self.timeit(number)
  File "/usr/lib/python3.6/timeit.py", line 178, in timeit
    timing = self.inner(it, self.timer)
  File "<timeit-src>", line 7, in inner
    str(choice(range(mini, maxi + 1)))
  File "/usr/lib/python3.6/random.py", line 258, in choice
    i = self._randbelow(len(seq))
OverflowError: Python int too large to convert to C ssize_t
$ python -m timeit 'from secrets import randbelow; length = 20' '"".join(str(randbelow(10)) for _ in range(length))'
10000 loops, best of 3: 103 usec per loop
```
On a donc une limite dans la taille des tokens qu'on peut générer avec ce code (contrairement à ma solution naïve qui est en `O(n)` mais qui s'adapte à toutes les longueurs).
D'un côté on peut se dire qu'on ne cherchera pas à faire des token de trop grande taille, mais d'un autre côté on peut aussi se dire que généré un token n'est pas vraiment quelque chose qui a besoin d'être optimisé et donc autant utilisé la solution naïve qui est la plus lisible...

Bref je te laisse seul juge